src/package-modules.js

```diff
@@ -113,6 +113,18 @@ function chooseNameAndVersion(instruction, package, magentoName, composerJson, d
   let composerConfig = JSON.parse(composerJson);
   let {version, name} = composerConfig;
   version = definedVersion || version || fallbackVersion;
+  
+  // If no version is found and ref looks like a branch name, convert to dev-branch format
+  if (!version && ref && !ref.match(/^v?\d+\.\d+/)) {
+    version = `dev-${ref}`;
+  }
```

Comment:
`ref` doesn't exist in this scope since the refactor. Need to update it to the new associated value. Maybe `instruction.ref` here? -- I don't remember the nightly code in detail, top of mind.